pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -174,6 +227,15 @@ func (c *objectCache) key(namespace, name string) string {
 	return name
 }
 
+func (c *objectCache) updateItem(key objectKey, item *objectCacheItem) *objectCacheItem {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	newItem := c.newReflector(key.namespace, key.name)
+	newItem.refCount = item.refCount
```

Comment:
See the comment below about.

I think this can probably be mitigated by simply calling Run() on the already existing reflector (which will simplify the whole code too).